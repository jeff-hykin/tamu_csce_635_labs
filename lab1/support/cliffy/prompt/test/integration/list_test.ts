import { List } from "../../list.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";

await snapshotTest({
  name: "list prompt",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should enter some keywords": { stdin: ["foo,bar", "\n"] },
  },
  async fn() {
    await List.prompt({
      message: "Enter some keywords",
      default: ["foo", "far"],
      hint: "some hint...",
    });
  },
});
