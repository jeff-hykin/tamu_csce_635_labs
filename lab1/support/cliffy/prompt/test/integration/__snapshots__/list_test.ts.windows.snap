export const snapshot = {};

snapshot[`list prompt > should enter some keywords 1`] = `
stdout:
"? Enter some keywords (foo, far) » 
❯ some hint...\\x1b[1A\\x1b[36G\\x1b[G\\x1b[0J
? Enter some keywords (foo, far) » foo, bar
❯ some hint...\\x1b[1A\\x1b[44G\\x1b[G\\x1b[0J
? Enter some keywords (foo, far) » foo, bar
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
